### ISSUE TITLE
Checkbox component does not complain with WAI ARIA

### ISSUE THREAD

**Issue by FilipePfluck** [chakra-ui/ark#3014]
### Description

Doesn't have role "checkbox"
Doesn't have aria-checked

### Link to Reproduction (or Detailed Explanation)

https://ark-ui.com/react/docs/components/checkbox

### Steps to Reproduce

inspect elements and look the checkbox html

### Ark UI Version

4.1.12

### Framework

- [ ] React
- [ ] Solid
- [ ] Vue
